restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/BaseVoiceInterpreter.java

```diff
@@ -1637,6 +1725,25 @@ public void execute(final Object message) throws Exception {
         }
     }
 
+    final class PartialGathering extends CallbackGatherAction {
+
+        public PartialGathering(final ActorRef source) {
+            super(source);
+        }
+
+        @Override
+        public void execute(final Object message) throws Exception {
+            final MediaGroupResponse<CollectedResult> asrResponse = (MediaGroupResponse<CollectedResult>) message;
+
+            final List<NameValuePair> parameters = parameters();
+            parameters.add(new BasicNameValuePair("UnstableSpeechResult", asrResponse.get().getResult()));
```

Comment:
Name of this parameter *UnstableSpeechResult* is confusing. we should make this *PartialSpeechResult*.
Also we can declare it as private static final string.